experimental/examples/bubblebench/editable-shared-tree/src/schema.ts

```diff
@@ -0,0 +1,234 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    brand,
+    EditableTree,
+    emptyField,
+    EmptyKey,
+    FieldKinds,
+    fieldSchema,
+    JsonableTree,
+    namedTreeSchema,
+    rootFieldKey,
+    SchemaData,
+    ValueSchema,
+} from "@fluid-internal/tree";
+
+export const stringSchema = namedTreeSchema({
+    name: brand("String"),
+    extraLocalFields: emptyField,
+    value: ValueSchema.String,
+});
+
+export const numberSchema = namedTreeSchema({
+    name: brand("number"),
+    extraLocalFields: emptyField,
+    value: ValueSchema.Number,
+});
+
+export const iBubbleSchema = namedTreeSchema({
+    name: brand("Test:BubbleBenchAppStateiBubble-1.0.0"),
+    localFields: {
+        x: fieldSchema(FieldKinds.value, [numberSchema.name]),
+        y: fieldSchema(FieldKinds.value, [numberSchema.name]),
+        r: fieldSchema(FieldKinds.value, [numberSchema.name]),
+        vx: fieldSchema(FieldKinds.value, [numberSchema.name]),
+        vy: fieldSchema(FieldKinds.value, [numberSchema.name]),
+    },
+    extraLocalFields: emptyField,
+});
+
+export const iBubbleSequenceSchema = namedTreeSchema({
+    name: brand("Test:BubbleBenchAppStateiBubbleSequence-1.0.0"),
+    localFields: {
+        [EmptyKey]: fieldSchema(FieldKinds.sequence, [iBubbleSchema.name]),
+    },
+    extraLocalFields: emptyField,
+});
+
+export const iClientSchema = namedTreeSchema({
+    name: brand("Test:BubbleBenchAppStateiClient-1.0.0"),
+    localFields: {
+        clientId: fieldSchema(FieldKinds.value, [stringSchema.name]),
+        color: fieldSchema(FieldKinds.value, [stringSchema.name]),
+        bubbles: fieldSchema(FieldKinds.sequence, [iBubbleSequenceSchema.name]),
+    },
+    extraLocalFields: emptyField,
+});
+
+export const iClientSequenceSchema = namedTreeSchema({
+    name: brand("Test:BubbleBenchAppStateiClientSequence-1.0.0"),
+    localFields: {
+        [EmptyKey]: fieldSchema(FieldKinds.sequence, [iClientSchema.name]),
+    },
+    extraLocalFields: emptyField,
+});
+
+export const AppStateSchema = namedTreeSchema({
+    name: brand("Test:BubbleBenchAppState-1.0.0"),
+    localFields: {
+        clients: fieldSchema(FieldKinds.sequence, [iClientSequenceSchema.name]),
+    },
+    extraLocalFields: emptyField,
+});
+
+export type BubbleTreeProxy = EditableTree & {
```

Comment:
To improve clarity for anyone who looks at this as an example use of tree:

```suggestion
// TODO: Generate this from schema automatically instead of hand coding it.
export type BubbleTreeProxy = EditableTree & {
```

Add that comment to the two types below as well.